orttraining/orttraining/python/training/optim/_ds_modifier.py

```diff
@@ -39,14 +39,22 @@ def can_be_modified(self):
         # it's safe to update the version supporting list. Otherwise, or the file is moved or renamed,
         # we need to check the implementation of these functions in detail.
         ds_version = Version(deepspeed.__version__)
-        if ds_version > Version("0.7.3") or ds_version < Version("0.4.0"):
+        if ds_version > Version("0.8.0") or ds_version < Version("0.4.0"):
             warnings.warn(
                 "Skip modifying optimizer because of unsupported DeepSpeed version {}, "
                 "supported version: 0.4.0 - 0.7.3.".format(deepspeed.__version__),
                 UserWarning,
             )
             return False
 
+        try:
+            from deepspeed.accelerator import get_accelerator
+        except ImportError as e:
+            pass
+        else:
+            if not get_accelerator().device_name().startswith("cuda"):
+                return False
```

Comment:
For better debugging, please also add a warning.warn here to specify the reason of failure.

Reply:
If possible, let's avoid warning unless it is really a warning. For debugging, we should probably add logging capabilities and log as debug or info.

Reply:
I think it's kind of warning or error. When it goes there, it means user requires to modify the optimizer, if we cannot do that, we should tell/warn user the reason.